app/src/main/java/acr/browser/lightning/settings/fragment/GeneralSettingsFragment.kt

```diff
@@ -330,32 +329,38 @@ class GeneralSettingsFragment : AbstractSettingsFragment() {
 
     private fun userAgentSummary() = choiceToUserAgent(userPreferences.userAgentChoice) + activity?.application?.let { ":\n" + userPreferences.userAgent(it) }
 
-    private fun choiceToUserAgent(index: Int) = when (index) {
-        1 -> resources.getString(R.string.agent_default)
-        2 -> resources.getString(R.string.agent_windows_desktop)
-        3 -> resources.getString(R.string.agent_linux_desktop)
-        4 -> resources.getString(R.string.agent_macos_desktop)
-        5 -> resources.getString(R.string.agent_android_mobile)
-        6 -> resources.getString(R.string.agent_ios_mobile)
-        7 -> resources.getString(R.string.agent_system)
-        8 -> resources.getString(R.string.agent_web_view)
-        9 -> resources.getString(R.string.agent_custom)
-        10 -> resources.getString(R.string.agent_hide_device)
+    private fun choiceToUserAgent(agent: String) = when (agent) {
```

Comment:
Not that it matters much but could we not use statically declared map of resource ids?

Reply:
Something like this could remove the need for having this in two places: one array for the order, and this function for mapping to resource string.

`LinkedHashMap` keeps the entries [in insertion order](https://stackoverflow.com/questions/2923856/is-the-order-guaranteed-for-the-return-of-keys-and-values-from-a-linkedhashmap-o), so we could use this map and create the array from the map's keys.

[and `LinkedHashMap` is what should be used in the third party cache of the blocker, I just didn't know about it until now]